src/gluonts/mx/distribution/ISQF.py

```diff
@@ -0,0 +1,864 @@
+            )
+
+        # Add tol to prevent the y-distance of two quantile knots from being too small
+        # Because in this case the spline knots could be squeezed together
+        # and cause overflow in spline CRPS computation
+        q_proj = F.concat(
+            F.slice_axis(q, axis=-1, begin=0, end=1),
+            F.abs(F.slice_axis(q, axis=-1, begin=1, end=None)) + tol,
+            dim=-1,
+        )
+        q_proj = cumsum(F, q_proj)
+
+        # Prevent overflow when we compute 1/beta
+        beta_l, beta_r = (
+            F.abs(beta_l.squeeze(axis=-1)) + tol,
+            F.abs(beta_r.squeeze(axis=-1)) + tol,
+        )
+
+        return knots, heights, q_proj, beta_l, beta_r
+
+    def distribution(
+        self,
+        distr_args,
+        loc: Optional[Tensor] = None,
+        scale: Optional[Tensor] = None,
+    ) -> ISQF:
+        """
+        function outputing the distribution class
+        distr_args: distribution arguments
+        loc: shift to the data mean
+        scale: scale to the data
+        """
+
+        distr_args, alpha = self.reshape_spline_args(distr_args, self.alpha)
+
+        if scale is None:
+            return self.distr_cls(
+                *distr_args, alpha, self.num_qk, self.num_pieces, self.tol
+            )
+        else:
+            distr = self.distr_cls(
+                *distr_args, alpha, self.num_qk, self.num_pieces, self.tol
+            )
+            return TransformedISQF(
+                distr, [AffineTransformation(loc=loc, scale=scale)]
+            )
+
+    def reshape_spline_args(self, distr_args, alpha):
+        """
+        auxiliary function reshaping
+        knots and heights to (*batch_shape, num_qk-1, num_pieces)
+        alpha to (*batch_shape, num_qk)
+        """
+
+        knots, heights = distr_args[0], distr_args[1]
+        beta_l = distr_args[3]
+        # can be deleted after reshape issue has been resolved
+        q = distr_args[2]
+        F = getF(beta_l)
+
+        # FIXME number 1
+        # Convert alpha from list of len=num_qk to Tensor of shape (*batch_shape, num_qk)
+        # For example, if alpha = [0.1, 0.5, 0.9],
+        # then alpha_reshape will be a Tensor of shape (*batch_shape, 3) with the last dimension being [0.1, 0.5, 0.9]
+        # In PyTorch, it would be torch.tensor(alpha).repeat(*batch_shape,1)
+        alpha_reshape = F.concat(
+            *[
+                F.expand_dims(F.ones_like(beta_l), axis=-1) * alpha[i]
+                for i in range(self.num_qk)
+            ],
+            dim=-1,
+        )
+
+        # FIXME number 2
+        # knots and heights have shape (*batch_shape, (num_qk-1)*num_pieces)
+        # I want to convert the shape to (*batch_shape, (num_qk-1), num_pieces)
+        # Here I make a shape_holder with the target shape, and use reshape_like
+
+        # create a shape holder of shape (*batch_shape, num_qk-1, num_pieces)
+        shape_holder = F.slice_axis(q, axis=-1, begin=0, end=-1)
```

Comment:
Do you have another array of the desired shape, like `heights`? Then you should be able to multiply the values of `q` by F.ones_like(height)`.

Reply:
Combine the following 3 lines into 1